### ISSUE TITLE
Frequent errors/crashes or deadlocks with the eye-dropper and saving in the 1.2 series

### ISSUE THREAD

**Issue by nekohayo** [mypaint/mypaint#507]
Same hardware and config as in issue #506 

I've noticed over the past few days of use that it is fairly easy to end up in some sort of deadlock situation with the app (where the OS/toolkit will declare it "unresponsive") when you use the eye dropper tool. It's not 100% of the time like in issue #506 though. I've been using a bunch of layers, the colour picker and eye-dropper, and the set no.1 penbrush.

The lockups have made me take the habit of doing ctrl+S _very_ frequently now, especially as the longer I go without saving the more likely it would be that saving would also stop working, either freezing the application, or "a programming error has been detected" would occur. This has made me a little bit nervous, to say the least :)

The whole deal is not really deterministic, I haven't figured out a solid pattern here. I know that's not terribly precise when it comes to finding the cause, but hopefully this helps a little.

I doubt that an .ora file weighting around a 100-170 megabytes (24x36" at 300 dpi) would be "too much" for MyPaint, right? :)

**Comment by achadwick** [comment]
Cannot reproduce this one.

What version are you running here? The regular eye dropper tools have changed quite recently in the 1.2.0-beta development.

Our eye dropper tools (footer or toolbar) should be *much* more efficient in recent MyPaint betas than before, at the loss of a small amount of functionality. Basically it's doing the same kind of render as the layer display, and only using the tiles near the cursor. In theory...

According to the other issue, you are using "Windows 7 64-bits (for the time being as I don't have a build for Fedora 23) with a Wacom Cintiq". Is this still correct? Are you still seeing the problem?

Does MyPaint appear to have a grab active? In other words, can you click buttons while this is happening?

Is MyPaint fullscreened when this happens?

Which eye dropper tool are you using? We have about 3 at the last count, one of which lives only in the legacy GTK widgets in the background selector.

**Comment by nekohayo** [comment]
Yep, that's with the Win7x64 beta2 and beta3 version, with the Cintiq. Happened to me quite a lot when I was drawing on Friday and Monday. I was not running fullscreened during those days.

The eye-dropper I'm referring to is the button that is to the absolute bottom-left of the main window, at the corner of the screen. It might be (in part?) issue #321 meddling and confusing my bug hunting. In this case I would (all with the stylus) click the button, pick the color in the canvas, then switch to the penbrush and continue drawing. Often it would work, and sometimes it wouldn't. 

…but I'm pretty sure I experienced many cases where I would hit the Save button on the toolbar and the app would go unresponsive (according to Windows) and the .ora file would indeed not get overwritten (probably a good thing ;) — it annoys me that I haven't found a really solid pattern to trigger it.

**Comment by achadwick** [comment]
[MyPaint v1.2.1](https://github.com/mypaint/mypaint/releases/tag/v1.2.1) was released recently. Please could you test it out, and let us know whether the bug is still happening?

(Sorry for the generic message, just encouraging people to retest. But it doesn't seem that this one is super reproducible either. I'm considering closing it unless there's a sure-fire way of tickling it.)
